source/common/router/config_impl.cc

```diff
@@ -640,14 +685,53 @@ RouteEntryImplBase::RouteEntryImplBase(const VirtualHostImpl& vhost,
   }
 
   if (route.route().has_regex_rewrite()) {
-    if (!prefix_rewrite_.empty()) {
-      throw EnvoyException("Cannot specify both prefix_rewrite and regex_rewrite");
+    if (!prefix_rewrite_.empty() || path_match_policy_.enabled()) {
+      throw EnvoyException("Specify only one of prefix_rewrite, regex_rewrite or path_rewrite_policy");
     }
     auto rewrite_spec = route.route().regex_rewrite();
     regex_rewrite_ = Regex::Utility::parseRegex(rewrite_spec.pattern());
     regex_rewrite_substitution_ = rewrite_spec.substitution();
   }
 
+  if (path_match_policy_.enabled()) {
+    if (!prefix_rewrite_.empty() || route.route().has_regex_rewrite()) {
+      throw EnvoyException("Specify only one of prefix_rewrite, regex_rewrite or path_rewrite_policy");
+    }
+  }
+
+  if (path_match_policy_.enabled()) {
+    // TODO: validate path_match_policy is of kind the pattern template
```

Comment:
This shouldn't be necessary -- the extension mechanism should allow other types of match policies.

Additional validation should happen in the specific extension's factory when the policy's implementation is constructed from its proto config, after the `downcastAndValidate` call in the factory.

Reply:
changed